File: application/src/Caravelo/Bundle/ApiBundle/BusinessCase/GetSurveyBusinessCase.php
<?php

namespace Caravelo\Bundle\ApiBundle\BusinessCase;


use Caravelo\Bundle\ApiBundle\Model\SurveyDto;
use Caravelo\Modules\Survey\Domain\Service\SurveyService;

class GetSurveyBusinessCase
{
    /**
     * @var SurveyService
     */
    private $surveyService;

    public function __construct(SurveyService $surveyService)
    {
        $this->surveyService = $surveyService;
    }

    /**
     * @param $surveySlug
     * @return bool|SurveyDto
     */
    public function getSurvey($surveySlug)
    {
        $survey = $this->surveyService->getSurvey($surveySlug);

        if($survey == false)
        {
            return false;
        }

        $surveyDto = new SurveyDto();
        $surveyDto->parseSurvey($survey);

        return $surveyDto;
    }
}

File: application-java/src/main/java/bussiness_case/GetSurveyBusinessCase.java
package bussiness_case;

import dto.SurveyDto;
import factory.SurveyDtoFactory;
import survey.domain.model.Survey;
import survey.domain.service.SurveyService;

public class GetSurveyBusinessCase {
	
	private SurveyService surveyService;
	private SurveyDtoFactory surveyDtoFactory;
	
	public GetSurveyBusinessCase(
			SurveyService surveyService,
			SurveyDtoFactory surveyDtoFactory){
		this.surveyService = surveyService; 
		this.surveyDtoFactory = surveyDtoFactory;
	}
	
	public SurveyDto getSurvey(String slug){
		
		Survey survey = this.surveyService.getSurvey(slug);
		
		return this.surveyDtoFactory.makeFromSurveyModel(survey);
	}
}
